optuna/samplers/_qmc.py

```diff
@@ -0,0 +1,357 @@
+                distributed optimization, all the samplers must share the same seed when the
+                `scrambling` is enabled. Otherwise, the low-discrepancy property of the samples
+                will be degraded.
+
+        search_space:
+            The search space of the sampler.
+
+            If this argument is not provided and there are prior
+            trials in the study, :class:`~optuna.samplers.QMCSamper` infers its search space using
+            the first trial of the study.
+
+            If this argument if not provided and the study has no
+            prior trials, :class:`~optuna.samplers.QMCSampler` samples the first trial using its
+            `_independent_sampler` and then infers the search space in the second trial.
+
+            .. note::
+                As mentioned above, the search space of the :class:`~optuna.sampler.QMCSampler` is
+                determined by argument ``search_space`` or the first trial of the study. Once
+                the search space is determined, it cannot be changed afterwards.
+
+        independent_sampler:
+            A :class:`~optuna.samplers.BaseSampler` instance that is used for independent
+            sampling. The first trial of the study and the parameters not contained in the
+            relative search space are sampled by this sampler.
+
+            If :obj:`None` is specified, :class:`~optuna.samplers.RandomSampler` is used
+            as the default.
+
+            .. seealso::
+                :class:`~optuna.samplers` module provides built-in independent samplers
+                such as :class:`~optuna.samplers.RandomSampler` and
+                :class:`~optuna.samplers.TPESampler`.
+
+        warn_independent_sampling:
+            If this is :obj:`True`, a warning message is emitted when
+            the value of a parameter is sampled by using an independent sampler.
+
+            Note that the parameters of the first trial in a study are sampled via an
+            independent sampler in most cases, so no warning messages are emitted in such cases.
+
+    Raises:
+        ValueError:
+            If ``qmc_type`` is not one of 'halton' an 'sobol`.
+
+    .. note::
+        Added in v2.x.0 TODO(kstoneriv3)as an experimental feature. The interface may change in
+        newer versions without prior notice.
+
+    Example:
+
+        Optimize a simple quadratic function by using :class:`~optuna.samplers.QMCSampler`.
+
+        .. testcode::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_float("x", -1, 1)
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+
+            sampler = optuna.samplers.QMCSampler()
+            study = optuna.create_study(sampler=sampler)
+            study.optimize(objective, n_trials=20)
+
+    """
+
+    def __init__(
+        self,
+        *,
+        qmc_type: str = "halton",
+        scramble: bool = False,
```

Comment:
This makes the sampling of shared QMC sequences on distributed workers simpler. Otherwise, one always has to specify a shared seed of QMC sampler at each worker in distributed settings. I guess we discussed this point a while ago.

Reply:
Sorry, you're right I found the explanation.